homeassistant/components/playstation_network/__init__.py

```diff
@@ -0,0 +1,78 @@
+"""The Playstation Network integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from psnawp_api.core.psnawp_exceptions import PSNAWPAuthenticationError
+from psnawp_api.psnawp import PSNAWP
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+
+from .const import DOMAIN, PLAYSTATION_NETWORK_API, PLAYSTATION_NETWORK_COORDINATOR
+from .coordinator import PlaystationNetworkCoordinator
+
+PLATFORMS: list[Platform] = [Platform.MEDIA_PLAYER]
+
+
+@dataclass
+class RuntimeData:
+    """Playstation Network Runtime Data."""
+
+    coordinator: PlaystationNetworkCoordinator
+
+
+type PlaystationNetworkConfigEntry = ConfigEntry[RuntimeData]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: PlaystationNetworkConfigEntry
+) -> bool:
+    """Set up Playstation Network from a config entry."""
+
+    try:
+        npsso: str = str(entry.data.get("npsso"))
+        psn = PSNAWP(npsso)
+    except PSNAWPAuthenticationError as error:
+        raise ConfigEntryAuthFailed(error) from error
+    except Exception as ex:
+        raise ConfigEntryNotReady(ex) from ex
+
+    try:
+        user = await hass.async_add_executor_job(lambda: psn.user(online_id="me"))
+        client = psn.me()
+        coordinator = PlaystationNetworkCoordinator(hass, psn, user, client)
+    except PSNAWPAuthenticationError as error:
+        raise ConfigEntryAuthFailed(error) from error
+    except Exception as ex:
+        raise ConfigEntryNotReady(ex) from ex
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
+        PLAYSTATION_NETWORK_COORDINATOR: coordinator,
+        PLAYSTATION_NETWORK_API: psn,
+    }
```

Comment:
I thought I removed all the references to this :)

I updated my other custom integration to use the new method but for this one I ran into an issue when defining a service call. I had the hass object but not the configEntry. I may have been doing something wrong, but since the docs said there were some exceptions to this method, I thought I may have had one.